### ISSUE TITLE
[Hold 12775][$1000] Workspace avatar turn into old avatar after being removed.

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#16728]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Go to any workspace, try to upload new avatar.
2. Try to remove it, click on avatar again to show the choose photo popup, wait for seconds
3. Notice that the Workspace avatar now return to the old one after being removed before that.

## Expected Result: 
Workspace avatar should stay empty.
#Actual Result: 
Workspace avatar now return to the old one after being removed before that.

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [x] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.2.92-0
**Reproducible in staging?:** y 
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/43996225/228704146-34fd2e0f-740d-4377-bc1a-120f3e48df84.MP4

https://user-images.githubusercontent.com/43996225/228704169-e2b635c6-d200-41d0-bb4b-e23210b6de80.MP4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @hungvu193 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1680106486925299

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01f8050b49fb096d4b</li>
        <li>Upwork Job ID: 1649935446442274816</li>
        <li>Last Price Increase: 2023-04-23</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @kevinksullivan (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by kevinksullivan** [comment]
I was unable to reproduce this @kavimuru . Let me know if I am missing something in my approach! 

https://user-images.githubusercontent.com/22508290/229202726-88894608-6937-40c0-8b99-7676b39ff8b3.MP4

**Comment by kavimuru** [comment]
@kevinksullivan hmm, I am not able to reproduce in the version 1.2.94-0

**Comment by kevinksullivan** [comment]
Going to close out for now then.

**Comment by MelvinBot** [comment]
@kevinksullivan  Be sure to fill out the [Contact List](https://forms.gle/zhdenCB5dE5gKfnw6)!

**Comment by hungvu193** [comment]
hey @kevinksullivan, I'm still able to reproduce, will send the video shortly

**Comment by hungvu193** [comment]
https://user-images.githubusercontent.com/16502320/229966540-209a96a8-176d-44dc-8b5d-1aa1e56bb93b.MOV

**Comment by kevinksullivan** [comment]
Hi @hungvu193 we tested on 1.2.94-0 most recently and couldn't reproduce. Can you test on a more up to date version?

**Comment by hungvu193** [comment]
@kevinksullivan sure, I'll wait for 1.2.96-4 version and let you know the result

**Comment by hungvu193** [comment]
@kevinksullivan I still can reproduce at latest ios version 1.2.97-2
https://user-images.githubusercontent.com/16502320/230757038-c22bc7af-3078-4916-a9c3-7f35d5599e9a.MOV

**Comment by hungvu193** [comment]
hey @kevinksullivan, I'm still able to reproduce this with IOS, mWeb Safari most recently on 1.3.0-0

**Comment by MelvinBot** [comment]
Triggered auto assignment to @sakluger (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by kevinksullivan** [comment]
I am OOO this week so reassigning for reproduction. Thanks @sakluger !

**Comment by MelvinBot** [comment]
@sakluger @kavimuru this issue is now 3 weeks old. There is one more week left before this issue <a href='https://stackoverflowteams.com/c/expensify/questions/15233/15234#15234'>breaks WAQ</a>. What needs to happen to get a PR in review this week? Please create a thread in #expensify-open-source to discuss. Thanks!

**Comment by sakluger** [comment]
I was able to reproduce using the MacOS native app on version 1.3.2-5. I also used the Take Photo option. After removing the avatar and clicking to add another, it reverted to the old avatar.

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~01f8050b49fb096d4b

**Comment by MelvinBot** [comment]
Current assignee @sakluger is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @sobitneupane (`External`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @amyevans (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by ahmedGaber93** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Workspace avatar turn into old avatar after being removed.
### What is the root cause of that problem?
This case is not implement, and we use default avatar image for both cases avater not set or avatar removed.
if there is no avatar url use default avatar
https://github.com/Expensify/App/blob/4947508bedc2e901082d805c84498f5f394ef655/src/pages/workspace/WorkspaceSettingsPage.js#L125
### What changes do you think we should make in order to solve the problem?
Soluation 1
when image update, we will add new attribute isUsingDefaultAvatar = false to policy object which mean the image is change before
```javascript
function updateWorkspaceAvatar(policyID, file) {
    const optimisticData = [{
        onyxMethod: CONST.ONYX.METHOD.MERGE,
        key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
        value: {
            isUsingDefaultAvatar: false, <<<<<<<<<<<<<<<
            ....
 ```           
 and where AvatarWithImagePicker used.
 note we also need to change this prop value in workspaces list and work space page.
 ```javascript
 <AvatarWithImagePicker
     isUsingDefaultAvatar={lodashGet(this.props.policy, 'isUsingDefaultAvatar', !Boolean(lodashGet(this.props.policy, 'avatar', false)))}
 ```
Soluation 2
when image removed, we can use avatar url = empty image url
### What alternative solutions did you explore? (Optional)

**Comment by hellohublot** [comment]
If the user finds that he has uploaded wrongly and wants to remove it, we probably should always display the default avatar (it's the original avatar)

If this is the desired behavior, https://github.com/Expensify/App/issues/15998#issuecomment-1479497357, or we can hold by #12775

**Comment by sobitneupane** [comment]
Thanks for the [proposal](https://github.com/Expensify/App/issues/16728#issuecomment-1518940296) @hellohublot. You have pointed out the root cause clearly. But I don't understand what you are trying to propose. Can you please elaborate the solution section of your proposal. Please use [the proposal template](https://github.com/Expensify/App/blob/main/contributingGuides/PROPOSAL_TEMPLATE.md). Propose your alternative solution in the designated place rather than listing solutions as A and B.

**Comment by hellohublot** [comment]
@sobitneupane Thank you for your suggestion. my solution is that after we delete it, we will no longer display the avatar we just uploaded.

I modified my [proposal](https://github.com/Expensify/App/issues/16728#issuecomment-1518940296), and you can also try the [code](https://github.com/Expensify/App/issues/15998#issuecomment-1479497357) here

**Comment by sobitneupane** [comment]
@hellohublot

> If the user finds that he has uploaded wrongly and wants to remove it, we probably should always display the default avatar (it's the original avatar)

The issue not only occurs when user removes the avatar. If user uploads two image in succession, he will be shown with the first one. Showing default avatar won't solve the issue. User should be shown with the latest changed image.

> We can use https://github.com/Expensify/App/issues/15998#issuecomment-1479497357

Can you please update your proposal to include the solution here. Also, do add the explanation about the changes you are suggesting.

> we can hold by https://github.com/Expensify/App/issues/12775

Do you mean we can hold this issue on behalf of the other issue https://github.com/Expensify/App/issues/12775.

**Comment by hellohublot** [comment]
@sobitneupane  Thanks

> The issue not only occurs when user removes the avatar. If user uploads two image in succession, he will be shown with the first one. Showing default avatar won't solve the issue. User should be shown with the latest changed image.

That is also caused by the `disordered socket message order`

**Comment by hellohublot** [comment]
>https://github.com/Expensify/App/blob/ebc03dbb14d1689e35301278833a82716ae53034/src/libs/actions/User.js#L287-L293

>We have received `{avatar: 'https://xxx'}` and `{avatar: null}` through the socket, this sequence order is correct, we should put them into onyx to update them

>But when we are waiting for the response of `SequentialQueue.getCurrenTrequest`, and the response of `UpdateWorkspaceAvatar` is later than `DeleteWorkspaceAvatar`, so the order of these two messages changed, and we finally updated to `{avatar: https://xxx}`


Hello @neil-marcellini. Sorry to bother you. May I ask if #12775 will fix `SequentialQueue.getCurrentRequest()` causing messages to be in the wrong order

**Comment by MelvinBot** [comment]
📣 @sobitneupane You have been assigned to this job by @amyevans!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) 📖

**Comment by amyevans** [comment]
Reassigned @sobitneupane as C+ since it looks like you were removed accidentally.

**Comment by neil-marcellini** [comment]
@hellohublot yes I think #12775 will fix this because we won't be sending any Push updates to the requesting client and will be sending them back in the HTTP responses instead. We can put this on hold, and once the replay effect fix implemented test to see if it's still happening.

**Comment by MelvinBot** [comment]
@sakluger @amyevans @sobitneupane this issue is now 4 weeks old and preventing us from maintaining WAQ, can you:
- Decide whether any proposals currently meet our guidelines and can be approved as-is today
- If no proposals meet that standard, please take this issue internal and treat it as one of your highest priorities
- If you have any questions, don't hesitate to start a discussion in #expensify-open-source

Thanks!

**Comment by MelvinBot** [comment]
Current assignee @sobitneupane is eligible for the Internal assigner, not assigning anyone new.

**Comment by sakluger** [comment]
Let's switch this to a weekly to put it in line with the Replay effect issue it's on hold for.

**Comment by sakluger** [comment]
Still on hold for that other issue.

**Comment by neil-marcellini** [comment]
Fixed! 

https://github.com/Expensify/App/assets/26260477/6585c081-fcfb-43d9-8f7e-349fb58c8ba2

**Comment by hungvu193** [comment]
@neil-marcellini am I egilible for reporting bonus?

**Comment by neil-marcellini** [comment]
Oh right sorry. @sakluger would you take care of @hungvu193 please.

**Comment by sakluger** [comment]
@neil-marcellini just to confirm, this was fixed by a PR for another issue? I'm pretty sure we decided that if another issue was already reported with the same root cause (the therefore the same PR would fix both), then we wouldn't pay for additional bug reports.

Did this bug report help to identify the root cause?

**Comment by neil-marcellini** [comment]
Yes it was fixed by a PR for the general replay effect tracking issue. This bug report did not help to identify the root cause, it was only another instance of the bug. https://github.com/Expensify/App/issues/12000 was the first issue where we identified the replay effect I believe.

Thanks for clarifying the policy that we don't pay for additional bug reports with the same root cause as another issue! That's why I leave the payment stuff to the payment experts like yourself 😄

There's plenty more bugs to report @hungvu193!
